### PR TITLE
sci-libs/umfpack: fix CI issue

### PR DESCRIPTION
**Pull request by waebbl**
- address the MissingUseDefault issue on sci-libs/umfpack from pkgcheck
- add src_compile function: needed to enable writing at /var/cache/fonts if USE=doc and not all pk fonts are available in the system by calls to mktexpk from pdflatex
- restrict dependencies on sci-libs/cholmod and sci-libs/amd

Closes: https://bugs.gentoo.org/716024

- sci-libs/cholmod: restrict dependencies on sci-libs/amd, sci-libs/colamd and sci-libs/parmetis
Closes: https://bugs.gentoo.org/716026
See also Bug: https://bugs.gentoo.org/682838

- dev-libs/zziplib: add pkg_setup function which is needed to properly setup a python implementation when USE=test is active.
Closes: https://bugs.gentoo.org/716028

Package-Manager: Portage-2.3.96, Repoman-2.3.22
Signed-off-by: Bernd Waibel <waebbl@gmail.com>

### CODE REVIEW

**Comment by orlitzky** [comment]
I added the new versions of the things he's blocking in cholmod/umfpack, so +1 from me on those changes.

For the `USE=doc` sandbox violation, I think we usually do `export VARTEXFONTS="${T}/fonts"` instead. (I'm not an expert here, but that's what e.g. the dev-lang/R ebuilds do).

**Comment by orlitzky** [comment]
> For the `USE=doc` sandbox violation, I think we usually do `export VARTEXFONTS="${T}/fonts"` instead. (I'm not an expert here, but that's what e.g. the dev-lang/R ebuilds do).

By the way, can you check to see if the latest version of umfpack needs this same fix?

**Comment by waebbl** [comment]
> > For the `USE=doc` sandbox violation, I think we usually do `export VARTEXFONTS="${T}/fonts"` instead. (I'm not an expert here, but that's what e.g. the dev-lang/R ebuilds do).
> 
> By the way, can you check to see if the latest version of umfpack needs this same fix?

I test whether the export works for this too. Would be better than writing to a location outside the sandbox. Will also check the current version for this.

**Comment by waebbl** [comment]
Using the export VARTEXFONTS method works. Thanks for the hint @orlitzky 
It's also needed for the current 5.7.9 ebuild.

File: sci-libs/umfpack/umfpack-5.6.2.ebuild [hunk 1]
```diff
@@ -14,11 +14,13 @@ SLOT="0"
...
 IUSE="doc +cholmod static-libs"
 
+# doesn't build against cholmod-3.0.13 and amd-2.4.6
 RDEPEND="
 	>=sci-libs/amd-1.3
+	<sci-libs/amd-2.4.6
 	sci-libs/suitesparseconfig
 	virtual/blas
-	cholmod? ( >=sci-libs/cholmod-2[-minimal] )"
+	cholmod? ( >=sci-libs/cholmod-2[-minimal(-)] <sci-libs/cholmod-3 )"
```
Comment (a17r) [inline sci-libs/umfpack/umfpack-5.6.2.ebuild:23]: this line makes no sense.

Comment (waebbl) [reply sci-libs/umfpack/umfpack-5.6.2.ebuild:23]: why? Should i just use the `<sci-libs/cholmod-3` term?

Comment (waebbl) [reply sci-libs/umfpack/umfpack-5.6.2.ebuild]: yes of course

**Comment by waebbl** [comment]
Thanks @a17r 
Revbumped the ebuilds, but not the last one (umfpack-5.7.9), this only added a src_compile function to fix building docs. Am I wrong, that there's no revbump needed?

**Comment by waebbl** [comment]
Has this changed? I thought it has been shown as a warning before

File: sci-libs/cholmod/cholmod-2.1.2-r1.ebuild [hunk 2]
```diff
@@ -14,24 +14,32 @@ SLOT="0"
...
 	doc? ( virtual/latex-base )"
 
+PATCHES=(
+	"${FILESDIR}/${P}-0001-Gentoo-specific-fix-self-linking-of-files.patch"
+)
+
 src_prepare() {
 	# bug #399483 does not build with parmetis-3.2
 	has_version "=sci-libs/parmetis-3.2*" && \
 		epatch "${FILESDIR}"/${PN}-1.7.4-parmetis32.patch
```
Comment (a17r) [inline sci-libs/cholmod/cholmod-2.1.2-r1.ebuild:40]: Don't use `epatch` in >=EAPI-6. Either replace with `eapply`, or add it to `PATCHES` in that condition.

But then it appears this whole block can be dropped now anyway, as <sci-libs/parmetis-4 is gone?

**Comment by orlitzky** [comment]
> Does it change the install image when building docs had failed before?

If building the docs worked for you before, nothing has changed. If building them didn't work, the difference is that now emerge doesn't crash, so it's fine without a revbvump.

> But then it appears this whole block can be dropped now anyway, as <sci-libs/parmetis-4 is gone?

Yeah, that (conditional, ugh) call to epatch was already there. Regardless, this is an old old old version of cholmod, and I plan to get rid of it ASAP in favor of v3.0.13 which is part of a newer suitesparse and fixes a bunch of other assorted sci bugs.

**Comment by waebbl** [comment]
> > Revbumped the ebuilds, but not the last one (umfpack-5.7.9), this only added a src_compile function to fix building docs. Am I wrong, that there's no revbump needed?
> 
> Does it change the install image when building docs had failed before?

No it shouldn't. If you have the fonts already in /var/cache/fonts the build succeeds and installs the documentation. If you don't have, emerge quites with an error and nothing is installed at all. This was the case at least on my tests.

File: sci-libs/cholmod/cholmod-2.1.2-r1.ebuild (hunk 2 above)
Comment (waebbl) [reply sci-libs/cholmod/cholmod-2.1.2-r1.ebuild:40]: Was thinking, I already changed this. I'm gonna fix it.

**Comment by a17r** [comment]
But why do we fix an ancient sci-libs/umfpack ebuild instead of stabilising the newer one?

**Comment by waebbl** [comment]
I was asked whether I have time to look after some of the many CI issues which pop up at every PR. Those changes are follow-ups from fixing just one single CI issue on umfpack.
I decided to start fixing it, because the umfpack-5.6.2 which caused the issue was the stable one, while the current umfpack-5.7.9 is unstable.

**Comment by waebbl** [comment]
Seeing the list getting longer on every PR I'm opening, I begin thinking we should enforce a higher quality standard on updates.

**Comment by orlitzky** [comment]
> Seeing the list getting longer on every PR I'm opening, I begin thinking we should enforce a higher quality standard on updates.

The CI  reports a lot of things that aren't serious problems, but that you should try not to introduce. Problems in stable packages can be particularly hard to fix without restabilizing, or violating some other policy. Other times, you'll get blamed as the result of a change that someone else made (like probably happened here when I added the new version). In short: sometimes it's OK to ignore them, but unfortunately you don't get the benefit of the doubt when submitting a PR.

In this case, the previous versions of suitesparse are 7+ years old, so requiring all of this to go off without a hitch might just ensure that suitesparse is never updated again. Careful what you wish for =)

I'll try to handle the stabilization soon.

**Comment by a17r** [comment]
The list is also getting longer because CI is reporting ever more things. But stabilise newer version is a perfectly fine (and in fact preferred) solution if we know that already is CI green.

**Comment by waebbl** [comment]
> In this case, the previous versions of suitesparse are 7+ years old, so requiring all of this to go off without a hitch might just ensure that suitesparse is never updated again. Careful what you wish for =)

I'm not asking to remove all the old packages at once. I'm asking to thoroughly test the packages on updates. Something, which is rightfully expected from me, as proxy maintainer on every PR. So I'm assuming, this is the standard way and should be done so by devs as well. ;) All those fixes we're trivial to fix and trivial to catch.

What I'm asking for is just a tool which blocks merging a PR as long as it has CI issues, severe enough to pop up in the list.

> The list is also getting longer because CI is reporting ever more things. But stabilise newer version is a perfectly fine (and in fact preferred) solution if we know that already is CI green. In that case, putting together a package list for stabilisation is time well spent, rather than copying existing fixes to older (and stable, were invasive changes should not be made) ebuilds.

Good point, which I will keep in mind for future CI fixes.
But I don't know where I copied existing fixes to older ebuilds. O.o

**Comment by a17r** [comment]
> But I don't know where I copied existing fixes to older ebuilds. O.o

I'm not saying that in the technical sense, but logically, EAPI=7 bump and assorted fixes were already contained in the bumps to current ~arch versions.

**Comment by waebbl** [comment]
Oh, ok I see.

**Comment by orlitzky** [comment]
> I'm not asking to remove all the old packages at once. I'm asking to thoroughly test the packages on updates. Something, which is rightfully expected from me, as proxy maintainer on every PR. So I'm assuming, this is the standard way and should be done so by devs as well. ;) All those fixes we're trivial to fix and trivial to catch.

The CI reports things that shouldn't be fixed, so it'll never happen. A priori what it reports are not even agreed-upon problems... it's whatever mgorny says it should report today. You have to suffer as a proxy-maintainer because the proxy-maint team doesn't have enough manpower to sift through what's important and what's not, so it's easier (and admittedly often correct) to say "you have to fix these before we'll even look at your PR."

For packages that already have a maintainer, you can use bugzilla instead. It skips the CI, and doesn't store our imporant development history on a proprietary platform in violation of the social contract`</philosophy>`

**Comment by waebbl** [comment]
@orlitzky @a17r 
Closing this. The umfpack 5.6.2 is no longer in tree and no package explicitly depends on cholmod-2.1.2.
We should probably remove the old package, as there are 3.0.1{3,4} ebuilds in tree.

**Pull request closed without merging**